src/app/search/autocomplete.ts

```diff
@@ -253,44 +250,72 @@ const closingQuoteRegex = /\w["']$/;
  */
 function findLastFilter(
   query: string,
-  caretIndex: number
+  caretIndex: number,
+  searchConfig: SearchConfig
 ): {
   term: string;
   index: number;
 } | null {
   // TODO: maybe include non-whitespace after the caret?
   const queryUpToCaret = query.slice(0, caretIndex);
-  const ast = parseQuery(queryUpToCaret);
-
-  // Find all the "bare" keywords at the end of the query. For example if the query is:
-  // name:foo bar baz
-  // then the trailingKeywordStrings are 'bar baz'
-  const trailingKeywordArgs: string[] = [];
-  traverseAST(
-    ast,
-    (filter) => {
-      if (filter.type === 'keyword' && !/\s+/.test(filter.args)) {
-        trailingKeywordArgs.push(filter.args);
-      } else {
-        return false;
+
+  // Find the index where an incomplete filter starts. For example if the query is:
+  // name:"foo" bar baz
+  // then the open keywords are "bar baz"
+  let earliestIncompleteFilter = -1;
+  try {
+    // We can use the query lexer for this to scan through tokens in the query without parsing the whole AST.
+    for (const token of lexer(queryUpToCaret)) {
+      switch (token.type) {
+        case 'filter': {
+          if (
+            // Ignore complete quoted tokens, they're definitively finished.
+            !token.quoted &&
+            // Match either bare words ...
+            (token.keyword === 'keyword' ||
+              // ... or name:foo style keywords
+              (searchConfig.kvFilters[token.keyword]?.format === 'freeform' &&
```

Comment:
Just checking if the format has a specific value is incorrect since it can also be an array of multiple formats, `canonicalFilterFormats(filter).includes('freeform')` is preferable. This would affect the `inloadout` filter, which has formats `['simple', 'freeform']`